### ISSUE TITLE
Custom Tokens Defined using Regular Expressions

### ISSUE THREAD

**Issue by sree93hari** [mageran/node-xsyn#5]
Custom Tokens Defined using Regular Expressions is working weirdly when we define multiple tokens 

---

E.g.
Tokens : Tokens Token {% $$ = $1 + " " + $2; %}
        | Token {% $$ = $1; %}
        ;

Token : Token1 {% $$=$1+'Token1'; %}
         | Token2 {% $$=$1+'Token2'; %}
         ;

token $ Token1 : '[h][_][a-z]' ;
## token $ Token2 : '[m][_][a-z]' ;

Input: 
h_a
m_d
h_b

Expected Output :
h_aToken1 m_dToken2 h_bToken1

Actual Output:
h_aToken1 m_dToken1 h_bToken1

**Comment by mageran** [comment]
yes, this is a bug. Internally, the matches are done correctly, but the
token generation is not propagating this. Will try to fix it for the next
update.
--Matthias
